src/dune_engine/process.ml

```diff
@@ -460,6 +460,7 @@ end
 
 let run_internal ?dir ?(stdout_to = Io.stdout) ?(stderr_to = Io.stderr)
     ?(stdin_from = Io.null In) ~env ~purpose fail_mode prog args =
+  Dtemp.init_env ();
```

Comment:
Why do we have `init_env` here?
It seems strange to have dune process' environment be different based on whether or not it's run any external process.
I imagine if we're calling `init_env` at all, we should do so at the start of the build.
It would be good to document the purpose of this call, as well. (is it to handle some known cases where we don't use `Dtemp`, or is it just-in-case)

Reply:
> It seems strange to have dune process' environment be different based on whether or not it's run any external process.

Well, it was deemed to be crappy to create a temporary directory even if dune didn't need to use it.

>  I imagine if we're calling init_env at all, we should do so at the start of the build.

`Process.run` may be used outside of builds as well.

Reply:
How is this problem handled in Jenga?

Reply:
I think we don't change jenga's own TMPDIR env var at all. We always pass explicit `env` to subprocesses.
I only now noticed that below `env` is optional, which explains why you need `Dtemp.init_env ()` (for the `None` case, presumably). I think it's better to make it non-optional (do `Option.value ~default` where `default` is computed once), and then we won't need to putenv.

That said, I don't want to entangle the two changes too much. Feel free to add a comment summarizing this conversation and leave it as is.

Reply:
Yeah, indeed that seems better and in fact that's how it worked before. Then I talked to Jeremie and he seemed to prefer `putenv` for some reason. In retrospect, just setting the environment seems simpler and I think we should just do that. When Jeremie is back he can try to explain why he liked `putenv` so much.

Reply:
Maybe on some OSes specifying the environment explicitly is slower. 
(I don't know exactly how we do it in dune, but at least the execve systemcall on Linux needs an explicit environment, anyway, so I expect it to at least be very similar).